### ISSUE TITLE
Why there is no error handling on out-bound skip of `InputBuffer`?

### ISSUE THREAD

**Issue by SleepyBag** [microsoft/bond#1158]
Here is the code of `InputBuffer::Skip`:

```
    void Skip(uint32_t size)
    {
        if (size > _blob.length() - _pointer)
        {
            return;
        }

        _pointer += size;
    }
```

It can be seen that when the size is too large, it simply does nothing - neither throw an exception nor return a fail code. Is this behavior by design? Given that the `_pointer` member is protected, there is no way we can know if the skip success after calling it.

**Comment by chwarr** [comment]
Hmmm... It looks like this is an oversight. I think seeking beyond the end of the buffer should set the position to `_blob.length()` so that the next read will fail with EOF. Is this a fix you could [contribute](https://github.com/microsoft/bond/blob/master/CONTRIBUTING.md) with a unit test?

```cpp
void Skip(uint32_t size)
{
    if (size > _blob.length() - _pointer)
    {
        _pointer = _blob.length();
    }
    else
    {
        _pointer += size;
    }
}
```

**Comment by SleepyBag** [comment]
I don't think it the right behavior, either. The right behavior in my opinion should be letting caller know *there is* an error, by either an error code or an exception, instead of behaving unexpectedly - if there is no further call, the caller would never know such an error.

**Comment by chwarr** [comment]
Throwing EOF after setting `_pointer` would be fine too. Just throwing wouldn't "poison" the stream for the future.

**Comment by SleepyBag** [comment]
> Throwing EOF after setting _pointer would be fine too.
Yes, this is a good choice. I'll seek to contribute to it.
